model/buildlogger.go

```diff
@@ -400,3 +404,148 @@ type LogLine struct {
 	Timestamp time.Time
 	Data      string
 }
+
+// Logs describes a set of buildlogger logs, typically related by some
+// criteria.
+type Logs struct {
+	Logs      []Log `bson:"results"`
+	env       cedar.Environment
+	populated bool
+	timeRange util.TimeRange
+}
+
+// LogFindOptions describes the search criteria for the Find function on Logs.
+type LogFindOptions struct {
+	TimeRange util.TimeRange
+	Info      LogInfo
+	Limit     int
+}
+
+// Setup sets the environment for the logs. The environment is require for
+// functions on Logs.
+func (l *Logs) Setup(e cedar.Environment) { l.env = e }
+
+// IsNil returns if the logs are populated or not.
+func (l *Logs) IsNil() bool { return l.populated }
+
+// Find returns the logs matching the given search criteria. The environment
+// should not be nil.
+func (l *Logs) Find(ctx context.Context, opts LogFindOptions) error {
+	if l.env == nil {
+		return errors.New("cannot find with a nil env")
+	}
+
+	if opts.TimeRange.IsZero() || !opts.TimeRange.IsValid() {
+		return errors.New("invalid time range given")
+	}
+
+	l.populated = false
+	findOpts := options.Find()
+	if opts.Limit > 0 {
+		findOpts.SetLimit(int64(opts.Limit))
```

Comment:
isn't it viable/correct to just set it to 0 and remove the if statement, also make opts.Limit int64 rather than require the conversion.